src/string-bytes.c

```diff
@@ -125,3 +125,64 @@ UINT16 string_bytes_copy_tpm2b(TPM2B *dest, TPM2B *src) {
 
     return rval;
 }
+
+bool string_bytes_is_host_big_endian(void) {
+
+    uint32_t test_word;
+    uint8_t *test_byte;
+
+    test_word = 0xFF000000;
+    test_byte = (uint8_t *) (&test_word);
+
+    return test_byte[0] == 0xFF;
+}
+
```

Comment:
The routine you posted the comment closest to is a run time test to determine endianess. As far as the others, they could be declared as macros if I did a loop, and the loop counter should be small enough for the loop-unroller to unroll them (thus preserving lack of branches). But I am not too worried about those being macro'd up right now.